mycroft/skills/skill_loader.py

```diff
@@ -0,0 +1,235 @@
+        self._communicate_load_status()
+
+    def _prepare_for_load(self):
+        self.load_attempted = True
+        self.loaded = False
+        self.instance = None
+
+    def _skip_load(self):
+        log_msg = 'Skill {} is blacklisted - it will not be loaded'
+        LOG.info(log_msg.format(self.skill_id))
+
+    def _load_skill_source(self):
+        """Use Python's import library to load a skill's source code."""
+        # TODO: Replace the deprecated "imp" library with the newer "importlib"
+        module_name = self.skill_id.replace('.', '_')
+        main_file_path = os.path.join(self.skill_directory, SKILL_MAIN_MODULE)
+        try:
+            with open(main_file_path, 'rb') as main_file:
+                skill_module = imp.load_module(
+                    module_name,
+                    main_file,
+                    main_file_path,
+                    ('.py', 'rb', imp.PY_SOURCE)
+                )
+        except FileNotFoundError as f:
+            error_msg = 'Failed to load {} due to a missing file.'
+            LOG.error(error_msg.format(self.skill_id))
+            LOG.exception(f)
+        except Exception as e:
+            LOG.error("Failed to load skill: " + self.skill_id)
+            LOG.exception(e)
+
+        module_is_skill = (
+            hasattr(skill_module, 'create_skill') and
+            callable(skill_module.create_skill)
+        )
+        if module_is_skill:
+            return skill_module
+
+    def _create_skill_instance(self, skill_module):
+        """Use v2 skills framework to create the skill."""
+        self.instance = skill_module.create_skill()
+        self.instance.skill_id = self.skill_id
+        self.instance.settings.allow_overwrite = True
+        self.instance.settings.load_skill_settings_from_file()
+        self.instance.bind(self.bus)
+        try:
+            self.instance.load_data_files(self.skill_directory)
+            # Set up intent handlers
+            # TODO: can this be a public method?
+            self.instance._register_decorated()
+            self.instance.register_resting_screen()
+            self.instance.initialize()
+        except Exception:
+            # If an exception occurs, make sure to clean up the skill
+            self.instance.default_shutdown()
+            raise
+
+        self.loaded = True
+        self.last_loaded = time()
+
+    def _check_for_first_run(self):
+        """The very first time a skill is run, speak the intro."""
+        first_run = self.instance.settings.get(
+            "__mycroft_skill_firstrun",
+            True
+        )
+        if first_run:
+            LOG.info("First run of " + self.skill_id)
+            self.instance.settings["__mycroft_skill_firstrun"] = False
+            self.instance.settings.store()
+            intro = self.instance.get_intro_message()
+            if intro:
+                self.instance.speak(intro)
+
+    def _communicate_load_status(self):
```

Comment:
That would in essence pass the loaded value twice, once in the self argument and again as a separate argument.  Not sure I understand how this makes it more clear.  If I saw that, I would be confused as to why the value was passed as a parameter if it is available as an instance attribute.

Reply:
What I'm meaning is that it should be considered to be a return status / parameter instead of an instance attribute.

Reply:
are you asking me to call to this method in _load like this?
`self._communicate_load_status(self.loaded)`

Reply:
This is really minor and something that can be cleaned up later.

But I'd imagine `_load()` looking like this:

```python
    def _load(self):
        loaded = False
        self._prepare_for_load()
        if self.is_blacklisted:
            self._skip_load()
        else:
            skill_module = self._load_skill_source()
            if skill_module is not None:
                loaded = self._create_skill_instance(skill_module)
                self._check_for_first_run()
        self._communicate_load_status(loaded)
        self.loaded = True
```

And having` _create_skill_instance()` return the loaded status instead of setting the self.loaded member.

Just avoiding passing information invisibly between methods through members and sort of centralizing where status is being set.